### ISSUE TITLE
MLIR generated using saved/cached loss tensors instead of loss operand

### ISSUE THREAD

**Issue by raikonenfnu** [llvm/mlir-npcomp#139]
Aim: Tracing the computation graph of 1 iteration of training from pytorch to mlir using modulebuilder. 

Models tried: simple conv mnist model, dlrm, and a very simple 2 layer fully connected network.

Current status: Forward pass works great! but for the backward pass, it uses the loss tensors generated during the trace.

Way to spot:
We can see that the operand assigned to the loss computed is only used during it's assignment and return of function

Sample example:
Python script + mlir file generated is stored:
[simple_example.zip](https://github.com/llvm/mlir-npcomp/files/5675872/simple_example.zip)

Thanks yall :)

**Comment by silvasean** [comment]
Symptom seems similar to https://github.com/llvm/mlir-npcomp/issues/135

Likely same issue of having the tracing dispatch priorities wrong and not intercepting at the right point.
